apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -1464,33 +1485,41 @@ module.exports = {
                   elementDependencies.coreElementScripts = elementDependencies.elementScripts;
                   delete elementDependencies.elementScripts;
                 }
+                if (_.has(elementDynamicDependencies, 'elementScripts')) {
+                  elementDynamicDependencies.coreElementScripts =
+                    elementDynamicDependencies.elementScripts;
+                  delete elementDynamicDependencies.elementScripts;
+                }
               }
 
-              const dependencyTypes = [
-                'coreStyles',
-                'coreScripts',
-                'nodeModulesStyles',
-                'nodeModulesScripts',
-                'clientFilesCourseStyles',
-                'clientFilesCourseScripts',
-                'coreElementStyles',
-                'coreElementScripts',
-                'courseElementStyles',
-                'courseElementScripts',
-              ];
-              for (const type of dependencyTypes) {
-                if (_.has(elementDependencies, type)) {
-                  if (_.isArray(elementDependencies[type])) {
```

Comment:
I believe this test is not actually necessary as (1) it is not done a few lines above for question dependencies, and (2) if the dependencies are not an array they would fail the schema.